eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubClient.java

```diff
@@ -34,20 +34,22 @@
     private final AtomicBoolean hasConnection = new AtomicBoolean(false);
 
     //TODO (conniey): Can we remove this and replace with an TokenProvider?
-    private final ConnectionStringBuilder connectionStringBuilder;
+//    private final ConnectionStringBuilder connectionStringBuilder;
+    private final ConnectionParameters connectionParameters;
     //TODO (conniey): Replace with configured values in EventHubClientBuilder.
     private final Duration timeout = Duration.ofSeconds(45);
     private final String eventHubName;
 
-    EventHubClient(ConnectionStringBuilder connectionStringBuilder, TokenProvider tokenProvider,
-                   ReactorProvider provider, ReactorHandlerProvider handlerProvider, Scheduler scheduler) {
-        Objects.requireNonNull(connectionStringBuilder, "'connectionStringBuilder' is null");
-        Objects.requireNonNull(connectionStringBuilder.endpoint(), "'connectionStringBuilder.endpoint()' is null.");
 
-        this.connectionStringBuilder = connectionStringBuilder;
-        this.eventHubName = connectionStringBuilder.eventHubName();
-        this.host = connectionStringBuilder.endpoint().getHost();
+    EventHubClient(ConnectionParameters connectionParameters, ReactorProvider provider, ReactorHandlerProvider handlerProvider) {
+        Objects.requireNonNull(connectionParameters, "'connectionParameters' is null");
+        this.connectionParameters = connectionParameters;
+        this.eventHubName = connectionParameters.getCredentials().eventHubName();
```

Comment:
What about the case where the user does not pass in a SASToken with the "EntityPath" and sets it on the EventhubClientBuilder instead?

Reply:
Do you mean to change the constructor to include entityPath param, such as EventHubClient(ConnectionParameters connectionParameters, ReactorProvider provider, ReactorHandlerProvider handlerProvider, String entityPath) ?

Reply:
Do we have a default eventhubs name?

Reply:
No. If the CredentialsInfo does not contain eventHubName(), we should be using the one provided from the EventHubClientBuilder. To keep it consistent, lets rename CredentialInfo.eventHubName to CredentialInfo.eventHubPath.

There should be a check before creating an EventHubClient that:

```java
public class EventHubClientBuilder {
   public EventHubClientBuilder eventHubPath(String eventHubPath) {
        this.eventHubPath = eventHubPath;
        return this;
   }
 
   public EventHubClient build() {
       if (ImplUtils.isNullOrEmpty(credentials.getCredentials().eventHubName()) && ImplUtils.isNullorEmpty(this.eventHubPath()) {
           throw new IllegalStateException("eventHubPath is not set in credentials nor is it set on the builder.");
       }

       // other validations.
   }

```

Reply:
Moved EventHubPath to CredentialInfo